### ISSUE TITLE
createHistory() fails when the base route is not the domain root.

### ISSUE THREAD

**Issue by itsthatguy** [rackt/history#58]
When using nginx to serve your application at a url such as, `http://mydomain.com/some/base` `createHistory()` cannot determine the application root url.

You get the following error when trying to visit your application: 

```
Warning: Location "/some/base" did not match any routes
```

> same issue on react-router: https://github.com/rackt/react-router/issues/1960

**Comment by mjackson** [comment]
This is a router-specific issue. Let's follow up there.

**Comment by mjackson** [comment]
I added support for base URLs in 4adf71550d1f795ca2ad425fe3856eadb2da5843. Use it like:

```js
import { createHistory, useBasename } from 'history'

let history = useBasename(createHistory)({
  basename: '/some/base'
})
```

This API is released in 1.10.1

**Comment by itsthatguy** [comment]
Awesome! thanks @mjackson
